x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapperTests.java

```diff
@@ -536,6 +542,68 @@ private MapperService mapperServiceForFieldWithModelSettings(
         return mapperService;
     }
 
+    public void testExistsQuerySparseVector() throws IOException {
+        final String fieldName = "semantic";
+        final String inferenceId = "test_service";
+
+        MapperService mapperService = mapperServiceForFieldWithModelSettings(
+            fieldName,
+            inferenceId,
+            new SemanticTextField.ModelSettings(TaskType.SPARSE_EMBEDDING, null, null, null)
+        );
+
+        Mapper mapper = mapperService.mappingLookup().getMapper(fieldName);
+        assertNotNull(mapper);
+        SearchExecutionContext searchExecutionContext = createSearchExecutionContext(mapperService);
+        Query existsQuery = ((SemanticTextFieldMapper) mapper).fieldType().existsQuery(searchExecutionContext);
+        assertThat(existsQuery, instanceOf(ESToParentBlockJoinQuery.class));
+        ESToParentBlockJoinQuery toParentBlockJoinQuery = (ESToParentBlockJoinQuery) existsQuery;
+        SparseVectorFieldMapper sparseVectorFieldMapper = (SparseVectorFieldMapper) mapperService.mappingLookup()
+            .getMapper(getEmbeddingsFieldName(fieldName));
+        Query expectedQuery = NestedQueryBuilder.toQuery(
+            context -> sparseVectorFieldMapper.fieldType().existsQuery(context),
+            SemanticTextField.getChunksFieldName(fieldName),
+            toParentBlockJoinQuery.getScoreMode(),
+            false,
+            searchExecutionContext
+        );
+        assertThat(toParentBlockJoinQuery, equalTo(expectedQuery));
```

Comment:
I see what you mean. WDYT about something like this?

```suggestion
        assertThat(toParentBlockJoinQuery.getChildQuery(), instanceOf(BooleanQuery.class));
        BooleanQuery booleanQuery = (BooleanQuery) toParentBlockJoinQuery.getChildQuery();
        assertTrue(
            booleanQuery.clauses()
                .stream()
                .anyMatch(b -> b.getQuery().equals(sparseVectorFieldMapper.fieldType().existsQuery(searchExecutionContext)))
        );
```

We still need to know that we're creating a boolean query under the hood (which isn't entirely bad IMO), but we're duplicating less.

Reply:
I still find this very leaky. Given that we're already testing the query with the YAML tests, I'd opt for the simplest solution here, and just asserting the top `ESToParentBlockJoinQuery`.

Reply:
Fair enough. Thanks for iterating on this with me!

Reply:
It was a good idea Mike, thanks for sharing!